src/kit.ts

```diff
@@ -602,18 +571,38 @@ const VsGenerators: {[key: string]: string} = {
   16: 'Visual Studio 16 2019'
 };
 
-async function varsForVSInstallation(inst: VSInstallation, arch: string): Promise<Map<string, string>|null> {
-  console.log(`varsForVSInstallation path:'${inst.installationPath}' version:${inst.installationVersion} arch:${arch}`);
+async function varsForVSInstallation(inst: VSInstallation, hostArch: string, targetArch: string): Promise<Map<string, string>|null> {
+  console.log(`varsForVSInstallation path:'${inst.installationPath}' version:${inst.installationVersion} host arch:${hostArch} - target arch:${targetArch}`);
+
+  // TODO: if needed, cover also the scenario of VS older than 2017,
+  // which has vcvarsall.bat in a different location (Program Files x86\Microsoft Visual Studio [ver]\VC)
+  // and which is not recognizing x64 alias for amd64.
   const common_dir = path.join(inst.installationPath, 'Common7', 'Tools');
   let devbat = path.join(inst.installationPath, 'VC', 'Auxiliary', 'Build', 'vcvarsall.bat');
   const majorVersion = parseInt(inst.installationVersion);
   if (majorVersion < 15) {
     devbat = path.join(inst.installationPath, 'VC', 'vcvarsall.bat');
   }
-  const variables = await collectDevBatVars(devbat, [`${arch}`], majorVersion, common_dir);
+  const variables = await collectDevBatVars(devbat, [`${kitHostTargetArch(hostArch, targetArch)}`], majorVersion, common_dir);
   if (!variables) {
     return null;
   } else {
+    // When invoked for arm or arm64, the vcvarsall.bat script creates an x86 environment,
+    // when the arm(64) components are not installed.
+    // The way to differentiate between the two cases is to check that the first path
+    // in the PATH environment variable ends with the target architecture.
```

Comment:
Are you 100% sure this will always be the case forever?  I would hate to be inadvertently broken because msvc decided to change something without us knowing.

Reply:
No, not 100% sure. But I just noticed that a VS with arm/arm64 installed has some arm/arm64 specific vcvars, which call into the main one. I think that is a better indication.